lib/util/LibraryHelpers.js

```diff
@@ -0,0 +1,30 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+/** @typedef {import("../../declarations/WebpackOptions").LibraryOptions} LibraryOptions */
+/** @typedef {import("../../declarations/WebpackOptions").LibraryType} LibraryType */
+/** @typedef {import("../Chunk")} Chunk */
+/** @typedef {import("../Compilation")} Compilation */
+
+/**
+ * Determine library type from chunk entry options or compilation output options
+ * @param {Chunk} chunk The chunk to get library type for
+ * @param {Compilation} compilation The compilation
+ * @returns {LibraryType | undefined} The library type or undefined
+ */
+module.exports.getLibraryType = (chunk, compilation) => {
```

Comment:
Extracted the library type detection logic into a new helper module

Ref:
https://github.com/webpack/webpack/blob/613a5ada7e676573758e4714d2e345df81c1270d/lib/RuntimePlugin.js#L279-L284


I can also include the RuntimePlugin refactoring in this PR